src/bidi_class.rs

```diff
@@ -0,0 +1,145 @@
+    //     Noncharacter_Code_Point
+    //
+    // For all other cases:
+    //
+    //  All code points not explicitly listed for Bidi_Class
+    //  have the value Left_To_Right (L).
+    #[rustfmt::skip]
+    let default_class_assignments = [
+        (0x0600,0x07BF, "AL"), (0x0860,0x086F, "AL"), (0x08A0,0x08FF, "AL"), (0xFB50,0xFDCF, "AL"),
+        (0xFDF0,0xFDFF, "AL"), (0xFE70,0xFEFF, "AL"), (0x00010D00,0x00010D3F, "AL"),
+        (0x00010F30,0x00010F6F, "AL"), (0x0001EC70,0x0001ECBF, "AL"), (0x0001ED00,0x0001ED4F, "AL"),
+        (0x0001EE00,0x0001EEFF, "AL"),
+
+        (0x0590, 0x05FF, "R"), (0x07C0, 0x085F, "R"), (0x0870, 0x089F, "R"), (0xFB1D, 0xFB4F, "R"),
+        (0x00010800, 0x00010CFF, "R"), (0x00010D40, 0x00010F2F, "R"), (0x00010F70, 0x00010FFF, "R"),
+        (0x0001E800, 0x0001EC6F, "R"), (0x0001ECC0, 0x0001ECFF, "R"), (0x0001ED50, 0x0001EDFF, "R"),
+        (0x0001EF00, 0x0001EFFF, "R"),
+
+        (0x20A0, 0x20CF, "ET")
+    ];
+    // Collect the codepoints that may default to BN
+    let mut maybe_boundary_neutral = BTreeSet::new();
+    for x in &core_prop {
+        if &x.property == "Default_Ignorable_Code_Point"
+            || &x.property == "Noncharacter_Code_Point"
+        {
+            maybe_boundary_neutral
+                .extend(x.codepoints.into_iter().map(|c| c.value()));
+        }
+    }
+
+    // Process unassigned codepoints
+    let left_to_right_name = bidi_class_name("L")?;
+    let boundary_neutral_name = bidi_class_name("BN")?;
+    for cp in 0..=0x10FFFF {
+        if !assigned.contains(&cp) {
+            // Check if this code point is in the default Bidi classes
+            if let Some(class) =
+                lookup_unassigned(cp, &default_class_assignments)
+            {
+                let name = bidi_class_name(class)?;
+                by_type.get_mut(&name).unwrap().insert(cp);
+            } else if maybe_boundary_neutral.contains(&cp) {
+                by_type.get_mut(&boundary_neutral_name).unwrap().insert(cp);
+            } else {
+                // All others get assigned Left_To_Right
+                by_type.get_mut(&left_to_right_name).unwrap().insert(cp);
+            }
+        }
+    }
+
+    let mut wtr = args.writer("bidi_class")?;
+    if args.is_present("enum") {
+        wtr.ranges_to_enum(args.name(), &by_type)?;
+    } else if args.is_present("rust-enum") {
+        let variants = by_type.keys().map(String::as_str).collect::<Vec<_>>();
+        wtr.ranges_to_rust_enum(args.name(), &variants, &by_type)?;
+    } else {
+        wtr.names(by_type.keys())?;
+        for (name, set) in by_type {
+            wtr.ranges(&name, &set)?;
+        }
+    }
+
+    Ok(())
+}
+
+/// Look up a code point in the unassigned default Bidi classes.
+fn lookup_unassigned<'a>(
+    codepoint: u32,
+    defaults: &[(u32, u32, &'a str)],
+) -> Option<&'a str> {
+    defaults
+        .iter()
+        .find(|&&(start, end, _)| codepoint >= start && codepoint <= end)
```

Comment:
For range tests like this, I'd rather the style `start <= codepoint && codepoint <= end`, which I think flows a bit better.